pkg/database/repository.go

```diff
@@ -45,6 +51,9 @@ func (r repository) FindBySlug(slug string) (*model.Database, error) {
 		Preload("Lock").
 		Where("slug = ?", slug).
 		First(&d).Error
+	if errors.Is(err, gorm.ErrRecordNotFound) {
+		return nil, errdef.NewNotFound("database not found by slug: %s", slug)
```

Comment:
I would quote strings using `%q`. This string might not contain any whitespace but its probably an easier convention than to decide based on what string it is 🤔